src/agent/src/rpc.rs

```diff
@@ -807,25 +816,24 @@ impl protocols::agent_ttrpc::AgentService for AgentService {
             )
         })?;
 
-        if p.term_master.is_none() {
+        if let Some(fd) = p.term_master {
+            unsafe {
+                let win = winsize {
+                    ws_row: req.row as c_ushort,
+                    ws_col: req.column as c_ushort,
+                    ws_xpixel: 0,
+                    ws_ypixel: 0,
+                };
+
+                let err = libc::ioctl(fd, TIOCSWINSZ, &win);
+                Errno::result(err).map(drop).map_err(|e| {
+                    ttrpc_error(ttrpc::Code::INTERNAL, format!("ioctl error: {:?}", e))
+                })?;
+            }
+        } else {
             return Err(ttrpc_error(ttrpc::Code::UNAVAILABLE, "no tty".to_string()));
         }
```

Comment:
```suggestion
let term_master = p
    .term_master
    .ok_or_else(||  Err(ttrpc_error(ttrpc::Code::UNAVAILABLE, "no tty".to_string())))?;
```

followed by the existing code but with `term_master` replacing `p.term_master.unwrap()`.

Reply:
I don't follow. Maybe you could clarify or raise a PR after this one lands to resolve? My fix works so to avoid holding this PR from landing any longer, you might want to help with the effort and raise a PR or two?

Reply:
Eh, fair enough.